zson/zson.go

```diff
@@ -0,0 +1,26 @@
+package zson
+
+import (
+	"github.com/brimsec/zq/zng"
+)
+
+func Implied(typ zng.Type) bool {
+	switch typ.(type) {
+	case *zng.TypeOfInt64, *zng.TypeOfTime, *zng.TypeOfFloat64, *zng.TypeOfBool, *zng.TypeOfBytes, *zng.TypeOfString, *zng.TypeOfIP, *zng.TypeOfNet, *zng.TypeOfType:
+		return true
+	}
+	return false
+}
+
+func SelfDescribing(typ zng.Type) bool {
```

Comment:
```suggestion
// SelfDescribing(t) returns true if the type can be unambiguously recovered from literal values of this type.
func SelfDescribing(typ zng.Type) bool {
```

Reply:
I tried to add a doc string to both funcs here to see if I could, but now I'm thinking `Implied` would have the same documentation. Is the difference that Implied is for simple types and SelfDescribing is for all type?

Reply:
Great point.  I added this as a comment and included similar verbiage in the spec:
```
+// SelfDescribing returns true for types whose type name can be entirely derived
+// from its typed value, e.g., a record type can be derived from a record value
+// because all of the field names and type names are present in the value, but
+// an enum type cannot be derived from an enum value because not all the enumerated
+// names are present in the value.  In the former case, a decorated typedef can
+// use the abbreviated form "(= <name>)", while the letter case, a type def must use
+// the longer form "<value> (<name> = (<type>))".
 func SelfDescribing(typ zng.Type) bool {
```